src/test/java/io/camunda/testing/assertions/IncidentAssertTest.java

```diff
@@ -0,0 +1,449 @@
+      deployProcess(client, ProcessPackLoopingServiceTask.RESOURCE_NAME);
+      startProcessInstance(client, ProcessPackLoopingServiceTask.PROCESS_ID);
+
+      // when
+      final ActivateJobsResponse jobActivationResponse = activateSingleJob();
+
+      final ActivatedJob actual = jobActivationResponse.getJobs().get(0);
+      client
+          .newThrowErrorCommand(actual.getKey())
+          .errorCode(ERROR_CODE)
+          .errorMessage(ERROR_MESSAGE)
+          .send()
+          .join();
+
+      final IncidentAssert incidentAssert = assertThat(actual).extractLatestIncident();
+
+      // then
+      assertThatThrownBy(() -> incidentAssert.hasErrorMessage(WRONG_VALUE))
+          .isInstanceOf(AssertionError.class)
+          .hasMessage(
+              "Error message was not 'wrong value' but was 'Expected to throw an error event with the code 'error' with message 'error occurred', but it was not caught. No error events are available in the scope.' instead");
+    }
+
+    @Test
+    void testWasRaisedInProcessInstanceFailure() throws InterruptedException {
+      // given
+      deployProcess(client, ProcessPackLoopingServiceTask.RESOURCE_NAME);
+      final ProcessInstanceEvent processInstanceEvent =
+          startProcessInstance(client, ProcessPackLoopingServiceTask.PROCESS_ID);
+
+      // when
+      final ActivateJobsResponse jobActivationResponse = activateSingleJob();
+
+      final ActivatedJob actual = jobActivationResponse.getJobs().get(0);
+      client
+          .newThrowErrorCommand(actual.getKey())
+          .errorCode(ERROR_CODE)
+          .errorMessage(ERROR_MESSAGE)
+          .send()
+          .join();
+
+      final IncidentAssert incidentAssert = assertThat(actual).extractLatestIncident();
+
+      // then
+      assertThatThrownBy(() -> incidentAssert.wasRaisedInProcessInstance(-1))
+          .isInstanceOf(AssertionError.class)
+          .hasMessage(
+              "Incident was not raised in process instance -1 but was raised in %d instead",
+              processInstanceEvent.getProcessInstanceKey());
+    }
+
+    @Test
+    void testOccurredOnElementFailure() throws InterruptedException {
+      // given
+      deployProcess(client, ProcessPackLoopingServiceTask.RESOURCE_NAME);
+
+      final Map<String, Object> variables =
+          Collections.singletonMap(
+              ProcessPackLoopingServiceTask.TOTAL_LOOPS, "invalid value"); // will cause incident
+
+      // when
+      final ProcessInstanceEvent instanceEvent =
+          startProcessInstance(client, ProcessPackLoopingServiceTask.PROCESS_ID, variables);
+      /* will raise an incident in the gateway because VAR_TOTAL_LOOPS is a string, but needs to be an int */
+      final ActivateJobsResponse jobActivationResponse = activateSingleJob();
+      final ActivatedJob job = jobActivationResponse.getJobs().get(0);
+      client.newCompleteCommand(job.getKey()).send().join();
+
+      // TODO remove this
+      Thread.sleep(100);
```

Comment:
🔧 This construction happens a lot in these tests. It might be nice to extract this to a separate method, or even make a method for it in the testing utilities.